src/SignalR/docs/specs/TransportProtocols.md

```diff
@@ -197,3 +205,78 @@ When data is available, the server responds with a body in one of the two format
 If the `id` parameter is missing, a `400 Bad Request` response is returned. If there is no connection with the ID specified in `id`, a `404 Not Found` response is returned.
 
 When the client has finished with the connection, it can issue a `DELETE` request to `[endpoint-base]` (with the `id` in the query string) to gracefully terminate the connection. The server will complete the latest poll with `204` to indicate that it has shut down.
+
+## Ack Protocol
+
+The ack protocol primarily consists of writing and reading framing around the data being sent and received.
+All sends need to start with a 24 byte frame. The frame is 2 12 byte base64 encoded values. The first base64 value when decoded is the length of the payload being sent (minus the framing) as an int64 value. The second base64 value when decoded is the ack ID as an int64 of how many bytes have been received from the other side so far.
+
+The second part of the protocol is for when the transport ungracefully reconnects and uses the Ack IDs to get any data that might have been missed during the disconnect window. This will be described after showing the framing.
+
+### Framing
+
+Consider the following example:
+
+0x41 0x67 0x41 0x41 0x41 0x41 0x41 0x41 0x41 0x41 0x41 0x3d 0x48 0x51 0x41 0x41 0x41 0x41 0x41 0x41 0x41 0x41 0x41 0x3d 0x48 0x69
```

Comment:
IMO formatting: `(hex) 41 67 41 41 41  ...` is clearer, shorter, and easier to grok (also below)